src/tools/compiletest/src/runtest.rs

```diff
@@ -350,10 +350,13 @@ impl<'test> TestCx<'test> {
             }
         } else {
             if proc_res.status.success() {
-                self.fatal_proc_rec(
-                    &format!("{} test compiled successfully!", self.config.mode)[..],
-                    proc_res,
-                );
+                {
+                    self.error(&format!("{} test did not emit an error", self.config.mode));
+                    if self.config.mode == crate::common::Mode::Ui {
+                        println!("note: by default, ui tests are expected not to compile");
```

Comment:
```suggestion
                        eprintln!("note: by default, ui tests are expected not to compile");
```
use the same stream as the error above (Which I assume is stderr)

Reply:
it's not, oh compiletest